### ISSUE TITLE
GridSearchCandidateGenerator

### ISSUE THREAD

**Issue by shan760107** [deeplearning4j/deeplearning4j#8751]
#### Issue Description

Please describe our issue, along with:
I want to use GridSearchCandidateGenerator for hyperparameter tuning. 
I put LearningRate Param and layerSize Param to multilayerspace like below.
and parameter doesn't change at all.
I guess that there is bug at getCandiate method.

#### Version Information
* Deeplearning4j version: 1.0.0-beta6
* Platform information (OS, etc): win 10 

#### Additional Information
        ParameterSpace<Integer> layerSizeParam = new DiscreteParameterSpace<>(32, 48, 64);
	ParameterSpace<Double> learningRateParam = new DiscreteParameterSpace<>(0.005, 0.007, 0.01);

        MultiLayerSpace hyperParamaterSpace = new MultiLayerSpace.Builder()
	                .seed(seed)
	                .biasInit(1)
	                .l2(1e-4)
	                .updater(new NesterovsSpace(learningRateParam))
	                .addLayer(new DenseLayerSpace.Builder().nIn(numInputs).nOut(layerSizeParam)
	                    .weightInit(WeightInit.XAVIER)
	                    .activation(Activation.RELU)
	                    .build())
	                .addLayer(new DenseLayerSpace.Builder().nIn(layerSizeParam).nOut(layerSizeParam)
	                    .weightInit(WeightInit.XAVIER)
	                    .activation(Activation.RELU)
	                    .build())
	                .addLayer(new OutputLayerSpace.Builder()
                        .lossFunction(LossFunctions.LossFunction.MSE)
	                    .weightInit(WeightInit.XAVIER)
	                    .activation(Activation.SOFTMAX)
	                    .nIn(layerSizeParam).nOut(numOutputs).build())
	                .build();

        CandidateGenerator candidateGenerator = new GridSearchCandidateGenerator(hyperParamaterSpace, 30, GridSearchCandidateGenerator.Mode.Sequential, null);

        OptimizationConfiguration optimizationConfiguration = new OptimizationConfiguration.Builder()
                .candidateGenerator(candidateGenerator)
                .dataSource(ExampleDataSource.class,dataSourceProperties)
                .modelSaver(modelSaver)
                .scoreFunction(scoreFunction)
                .terminationConditions(conditions)
                .build();
	    
	    
        IOptimizationRunner runner = new LocalOptimizationRunner(optimizationConfiguration, new MultiLayerNetworkTaskCreator());
        //Uncomment this if you want to store the model.
        StatsStorage ss = new FileStatsStorage(new File("HyperParamOptimizationStats.dl4j"));
        runner.addListeners(new ArbiterStatusListener(ss));
        UIServer.getInstance().attach(ss);
        //runner.addListeners(new LoggingStatusListener()); //new ArbiterStatusListener(ss)
        runner.execute();

**Comment by AlexDBlack** [comment]
Thanks for reporting this issue.
It has been fixed here: https://github.com/KonduitAI/deeplearning4j/pull/292
The fix will be made available on snapshots within a few days at the latest.
Instructions for using snapshots can be found here: https://deeplearning4j.org/docs/latest/deeplearning4j-config-snapshots
